cupy/_environment.py

```diff
@@ -122,12 +129,25 @@ def _get_cuda_path():
     if nvcc_path is not None:
         return os.path.dirname(os.path.dirname(nvcc_path))
 
+    # Use nvrtc dll path
+    nvrtc_path = get_nvrtc_path()
+    if nvrtc_path is not None:
+        return os.path.dirname(os.path.dirname(nvrtc_path))
```

Comment:
This line does not work if the goal is to support NVRTC wheel (nvidia-cuda-nvrtc-cuXX), because the DLL is installed to a subdir under site-packages: On Linux it's nvidia/cuda_nvrtc/lib. So you need 3 dirname calls not 2. But even if you find the root dir nvidia this way it still would not work for CuPy to find other CUDA components installed via pip.

xref: https://github.com/cupy/cupy/issues/8384#issuecomment-2187833916